src/imgui/ImGuiSymbols.cc

```diff
@@ -189,15 +267,60 @@ void ImGuiSymbols::paint(MSXMotherBoard* /*motherBoard*/)
 
 		im::TreeNode("Symbols per file", ImGuiTreeNodeFlags_DefaultOpen, [&]{
 			auto drawFile = [&](const FileInfo& info) {
+				auto* file = symbolManager.findFile(info.filename);
 				im::StyleColor(!info.error.empty(), ImGuiCol_Text, getColor(imColor::ERROR), [&]{
 					auto title = strCat("File: ", info.filename);
 					im::TreeNode(title.c_str(), [&]{
 						if (!info.error.empty()) {
 							ImGui::TextUnformatted(info.error);
 						}
 						im::StyleColor(ImGuiCol_Text, getColor(imColor::TEXT), [&]{
+							auto arrowSize = ImGui::GetFrameHeight();
+							auto extra = arrowSize + 2.0f * style.FramePadding.x;
+							ImGui::SetNextItemWidth(ImGui::CalcTextSize("3-3").x + extra);
+						        ImGui::AlignTextToFramePadding();
+
+							// build slot-subslot preview string
+							std::string preview = file->slot ? (isSlotExpanded(motherBoard, *file->slot & 3) ? tmpStrCat(*file->slot & 3, '-', (*file->slot >> 2) & 3).c_str() : tmpStrCat(*file->slot & 3).c_str()) : tmpStrCat("-").c_str();
+
+				                        im::Combo(tmpStrCat("Slot##", info.filename).data(), preview.c_str(), [&]{
+								int psSs = 0;
+
+								// initial state
+								if (ImGui::Selectable("-", !file->slot)) {
+									file->slot = {};
+									for (auto& symbol: file->getSymbols()) {
+										symbol.slot = file->slot;
+									}
```

Comment:
This loop is repeated 3 times (here on line 291, and also on lines 302 and 311).
Maybe factor it out in a small lambda function, and call this helper 3 times;
```
auto setSlot = [&](std::optional<uint8_t>& newSlot) {
    file->slot = newSlot;
    for (auto& symbol : file->getSymbols()) {
        symbol.slot = newSlot;
    }
};
```